### PR TITLE
feature #79: created an admin roles flow

### LINKED ISSUE #79: [FEATURE] Create an admin roles flow
We need to add pages "Search roles", "Edit role", "Create role"

## Front-end part

Search roles:

- Sidebar that already implemented.
- Search panel:
  - Search input that searches by Display name of role
  - Filter "Role type", multichoice dropdown with an ability to search and filter in it
  - Sorting dropdown with variants "by create date", "by display name", "by weight"
  - Button for changing direction of sorting (asc/desc)
- Button "Create" that opens "Create role" page
- Table: 
  - Id of role (uuid, not upper cased as in design)
  - Display name (if null, write "Не встановлено")
  - Role type (CAPTAIN, ADMIN, MODERATOR, STUDENT, USER, enum from back-end)
  - Weight (int value)
  - Button "Rights" that opens "Search grants of role" page
  - Button "Edit" that opens "Edit role" page
  - Button "Delete" that opens pop-up to delete role
  - Below the role ability to choose page size and page (from MUI table component)

image

Edit role:

- Button "Cancel" to return to "Search roles" page
- Button "Delete" that opens Pop-up. After deleting return to "Search roles" page
- Button "Save" that saves changes. After saving return to "Search roles" page
- Input "Display name" (string input, 3 min, 32 max)
- Dropdown "Role type" ((CAPTAIN, ADMIN, MODERATOR, STUDENT, USER, enum from back-end)
- Weight (int, min 1, max 5000)
- Button "Go to rights" that implements the same logic as button "Rights" on "Search roles" page

image

Create role:

- Button "Cancel" to return to "Search roles" page
- Button "Save" that saves changes. After saving return to "Search roles" page
- Input "Display name" (string input, 3 min, 32 max)
- Dropdown "Role type" ((CAPTAIN, ADMIN, MODERATOR, STUDENT, USER, enum from back-end)
- Weight (int, min 1, max 5000)
- Button "Go to rights" that implements the same logic as button "Rights" on "Search roles" page

image

## Back-end part

- Endpoint to get roles with an ability to 
  - filter by:
    - Role type array
    - Display name string
  - and sort by:
    - creation date
    - display name
    - weight

- Endpoint to edit role by id (with restricted access):
  - display name (min 3, max 32)
  - role type
  - weight (int, min 1, max 5000)

- Endpoint to create role by id (with restricted access):
  - display name (min 3, max 32)
  - role type
  - weight (int, min 1, max 5000)

- Endpoint for deleting role (by id)

### PR DESCRIPTION
**Pull request by illiakroshka**
- Added sort by creation date
- Added min and max value for weight
- Added min and max length for display name
- Updated swagger documentation

closes #79

### CODE REVIEW

**Review by aim4ik11 — DISMISSED** [review]

File: fictadvisor-web/src/components/pages/admin/admin-roles/edit-roles-page/EditRolesPage.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            size={InputSize.MEDIUM}
+            type={InputType.DEFAULT}
+            showRemark={false}
+            label="Вага"
+            placeholder="Вага"
+          />
+        </Box>
+        <Box sx={{ maxWidth: '149px' }}>
+          <Link
+            href={`/admin/grants`}
```
Comment (igaryakqwe) [inline fictadvisor-web/src/components/pages/admin/admin-roles/edit-roles-page/EditRolesPage.tsx:133]: remove curly braces

File: fictadvisor-web/src/lib/api/role/RoleAPI.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    const { data } = await client.delete(
+      `/roles/${roleId}`,
+      getAuthorizationHeader(),
+    );
+    return data;
+  }
+
+  async create(body: CreateRoleBody) {
+    const { data } = await client.post(
+      `/roles`,
```
Comment (igaryakqwe) [inline fictadvisor-web/src/lib/api/role/RoleAPI.ts:46]: you doesn't need to use quotes like this

File: fictadvisor-web/src/components/pages/admin/admin-roles/edit-roles-page/EditRolesPage.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        toastError.displayError(e.response?.data.message);
+      }
+    }
+  };
+  return (
+    <>
+      <Box sx={styles.header}>
+        <CardHeader
+          title="Редагування"
+          subheader={`${role.id}`}
```
Comment (igaryakqwe) [inline fictadvisor-web/src/components/pages/admin/admin-roles/edit-roles-page/EditRolesPage.tsx:70]: remove string literal

File: fictadvisor-web/src/components/pages/admin/admin-roles/search-roles-page/components/roles-list/RolesList.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                  color={TagColorMapper[role.name]}
+                />
+              </TableCell>
+              <TableCell align="left" scope="row">
+                <Typography variant="body2">{role.weight}</Typography>
+              </TableCell>
+              <TableCell align="right" sx={styles.actionsWrapper}>
+                <Box sx={styles.actions}>
+                  <Link
+                    href={`/admin/grants`}
```
Comment (igaryakqwe) [inline fictadvisor-web/src/components/pages/admin/admin-roles/search-roles-page/components/roles-list/RolesList.tsx:115]: remove curly braces

File: fictadvisor-web/src/lib/api/role/RoleAPI.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+  // async editUser(userId: string, body: ChangeUserBody) {
+  //   const { data } = await client.patch(
+  //     `/users/${userId}`,
+  //     body,
+  //     getAuthorizationHeader(),
+  //   );
+  //   return data;
+  // }
+}
```
Comment (igaryakqwe) [inline fictadvisor-web/src/lib/api/role/RoleAPI.ts:70]: wtf

**Review by igaryakqwe — CHANGES_REQUESTED** [review]

**Review by igaryakqwe — APPROVED** [review]

**Pull request merged**